bat-utils/lib/extras-hapi.js

```diff
@@ -32,6 +33,31 @@ exports.domainCompare = (a, b) => {
   }
 }
 
+const constantTimeEquals = (a, b) => {
+  let mismatch = a.length !== b.length
+  if (mismatch) {
+    b = a
```

Comment:
would it be better to do `b = mismatch ? a : b`? there's probably a slight diff between runtime with equal length strings and nonequal due to the extra assignment as it stands.

idk if an assignment to self would be optimized out or not

Reply:
well, that's the problem with optimizers. sometimes they do the wrong thing for the right reasons.

Reply:
@BrendanEich may be the best to answer that question :). however, I don't think it's necessary to prevent a timing attack against length of input, so just `crypto.timingSafeEqual(Buffer.from(a), Buffer.from(b))` would be sufficient. if you really wanted to hide the token length, i think it's better to just HMAC the inputs and compare those (since they would then be equal length).

Reply:
makes sense. `timingSafeEqual` throws an error on unequal length inputs so will stick with the current code to ensure they are equal